File: app/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/UpdateUserRequest.java
package edu.cnm.deepdive.seesomethingabq.model.dto;

public class UpdateUserRequest {
  private String displayName;
  private String email;

  public UpdateUserRequest(String displayName, String email) {
    this.displayName = displayName;
    this.email = email;
  }
}

File: app/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/UpdateUserRequest.kt
package edu.cnm.deepdive.seesomethingabq.model.dto

/**
 * Request payload for updating user profile information.
 *
 * This DTO is used when sending profile update requests to the backend via
 * `PATCH /users/me`. Only the fields provided (non-null values) will be updated.
 * Fields left as `null` will be ignored by the backend and remain unchanged.
 *
 * @property displayName New display name for the user (nullable; optional).
 * @property email New email address for the user (nullable; optional).
 */
data class UpdateUserRequest(
  val displayName: String? = null,
  val email: String? = null
)
